gnome-initial-setup/gis-driver.c

```diff
@@ -911,31 +942,38 @@ window_realize_cb (GtkWidget *widget, gpointer user_data)
 }
 
 static void
-emit_startup_sound (GApplication * app, gpointer user_data)
+startup_sound_cb (EOSHackSoundClient *client, GAsyncResult *res,
+    GisDriver *driver)
 {
-  g_autoptr(GDBusProxy) proxy = NULL;
+  gchar *uuid;
   g_autoptr(GError) error = NULL;
-  const GDBusProxyFlags proxy_flags =
-    G_DBUS_PROXY_FLAGS_DO_NOT_LOAD_PROPERTIES |
-    G_DBUS_PROXY_FLAGS_DO_NOT_CONNECT_SIGNALS;
-
-  proxy = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SESSION,
-                                         proxy_flags,
-                                         NULL,
-                                         "com.endlessm.HackSoundServer",
-                                         "/com/endlessm/HackSoundServer",
-                                         "com.endlessm.HackSoundServer",
-                                         NULL, &error);
+  GisDriverPrivate *priv = gis_driver_get_instance_private (driver);
+
+  uuid = eos_hack_sound_client_play_finish (priv->sound_client, res, &error);
   if (error)
     {
-      g_printerr ("Cannot create proxy for 'com.endlessm.HackSoundServer': %s",
-          error->message);
+      g_printerr ("Cannot get the startup sound UUID: %s", error->message);
       return;
     }
+  priv->startup_sound_uuid = uuid;
+}
 
-  g_dbus_proxy_call (proxy,
-                     "PlaySound", g_variant_new ("(s)", "FBE/startup"),
-                     G_DBUS_CALL_FLAGS_NONE, -1, NULL, NULL, NULL);
+static void
+emit_startup_sound (GApplication * app, gpointer user_data)
+{
+  GisDriverPrivate *priv = gis_driver_get_instance_private (app);
+  eos_hack_sound_client_play (priv->sound_client, "FBE/startup",
+                              (GAsyncReadyCallback) startup_sound_cb,
+                              GIS_DRIVER (app));
+}
+
+static void
+emit_page_changed_sound (GisAssistant *assistant, GisDriver *driver)
+{
+  GisDriverPrivate *priv = gis_driver_get_instance_private (driver);
+  if (gis_assistant_get_current_page (assistant))
```

Comment:
I don't understand this check – is there not always a current page?

Reply:
Because for some reason `gis_assistant_get_current_page (assistant)` returns NULL when you are in the first page *for the first time*... and I don't want to emit the "page changed" sound on the first page when the FBE appears for the first time.

Reply:
I maintain that this deserves a comment.

Reply:
I forgot this... Is this comment okay
```
This check avoids the FBE/page-changed sound being emmited when the first page is loaded for the first time.
```
?

Reply:
If not, propose a comment please.

Reply:
Sounds fine to me!